SparkPost/2web2ui
# FE-1296 - Resolves events page bug in which invalid ranges caused an application crash

[FE-1296](https://sparkpost.atlassian.net/browse/FE-1296)

### What Changed

- Adds check for valid range before updated returned date options in the `parseSearch` message events helper
- Adds supporting test case

### How To Test

1. Go to `/reports/message-events` and update the URL to `/reports/message-events?from=2021-01-21T17%3A27%3A38Z&range=29days&to=2021-01-21T18%3A27%3A38Z`
2. Verify that the `range` in the URL changes to `hour` and the page doesn't crash
3. Set different, valid ranges through the UI and through the URL and verify they still work as expected (i.e., range is still passed correctly to the relevant network requests)

### To Do
